### PR TITLE
CALS-5603 ApplicantCard subcards to pureComponent 

### PR DESCRIPTION
**Pull request by ratneshraval**
#### Which User story does this relate to?
- https://osi-cwds.atlassian.net/browse/CALS-5603

#### Brief feature description
- Converting application state to immutable and ApplicantCard->subcards to pureComponent

### CODE REVIEW

File: app/javascript/rfa_forms/rfa01a_edit_view/index.js [hunk 1]
```diff
@@ -42,6 +42,7 @@ export default class Rfa01EditView extends React.Component {
       const countyValue = (this.props.user && this.props.user.county_code)
       this.state.application.application_county = this.props.countyTypes.find(countyType => countyType.id === parseInt(countyValue))
     }
+    this.state.application = Immutable.fromJS(this.state.application)
```
Comment (nasar-cambria) [inline app/javascript/rfa_forms/rfa01a_edit_view/index.js]: When I am trying to save any applicant it gives me an error: ```this.state.application.toJS is not a function```

**Comment by nasar-cambria** [comment]
There are few other errors in console.

File: app/javascript/rfa_forms/rfa01a_edit_view/applicantCard.jsx [hunk 2]
```diff
@@ -87,11 +86,10 @@ export default class ApplicantCard extends React.Component {
             idPrefix={idPrefix}
             phoneTypes={this.props.phoneTypes}
             applicant_id={this.props.index}
-            phones={checkArrayObjectPresence(this.props.applicantFields.phones) || undefined}
+            phones={this.props.applicantFields.get('phones') || undefined}
```
Comment (lauherkcambria) [inline app/javascript/rfa_forms/rfa01a_edit_view/applicantCard.jsx]: what does get function give us when phones is null or undefined?

**Comment by lauherkcambria** [comment]
Nice! this is a lot of stuff, i want to take a little bit more time to understand how immutable js functions work. for example, looks like we can replace the 

`this.props.applicantFields.get('phones') || undefined`

`this.props.applicantFields.get('phones',  undefined)`

which is just a little cleaner.

This also makes me wonder if there are other neat little clean tricks in the immutable js library we can leverage.

**Pull request merged**